File: LeetCode/1946. Largest Number After Mutating Substring/solution.cpp
class Solution {
public:
    string maximumNumber(string num, vector<int>& change) {
        int n = num.size(), i = 0;
        while (i < n && change[num[i] - '0'] <= num[i] - '0')
            i++;
        while (i < n && change[num[i] - '0'] >= num[i] - '0')
            num[i] = change[num[i] - '0'] + '0', i++;
        return num;
    }
};

// 作者：吴自华
// 链接：https://leetcode-cn.com/circle/discuss/zesmlZ/view/SqRxlN/
// 来源：力扣（LeetCode）
// 著作权归作者所有。商业转载请联系作者获得授权，非商业转载请注明出处。

File: LeetCode/1946. Largest Number After Mutating Substring/solution.py
class Solution:
    def maximumNumber(self, num: str, change: List[int]) -> str:
        res = [int(ch) for ch in num]
        mutation = False
        for i, x in enumerate(res):
            if mutation and change[x] < x:
                break
            if change[x] > x:
                res[i] = change[x]
                mutation = True
        return ''.join(str(ch) for ch in res)